### ISSUE TITLE
AYAB window closes during Python installation

### ISSUE THREAD

**Issue by chris007de** [AllYarnsAreBeautiful/ayab-desktop#241]
Vanessa writes:

> I’ve downloaded and am trying to install the latest version of the AYAB software.  The installation goes on for about 2 minutes and then  disappears.
> The box that pops up during the install attempt has a line item that says, “Install Python”.  I’m not sure how I should do this.  I’m pretty certain I installed it with the last version.  Can you please  advise how to continue the install?
> I’m using a Mac Power Book computer with the latest version of OS X.

![image](https://user-images.githubusercontent.com/1631996/30428614-8bbdd014-9954-11e7-8d29-2a18367db5bb.png)

I asked her to send the /tmp/ayab_install.log, but so far she was not able to find it.

**Comment by skateness** [comment]
Found install log file and emailed to Chris.  Thanks!

**Comment by chris007de** [comment]
Here's the content of ayab_install.log
```
Downloading readline-6.3.tar.gz...
-> https://pyenv.github.io/pythons/56ba6071b9462f980c5a72ab0023893b65ba6debb4eeb475d7a563dc65cafd43
Installing readline-6.3...
Installed readline-6.3 to /Users/VJ/.pyenv/versions/3.5.0

Downloading Python-3.5.0.tar.xz...
-> https://www.python.org/ftp/python/3.5.0/Python-3.5.0.tar.xz
Installing Python-3.5.0...

BUILD FAILED (OS X 10.12.6 using python-build 20160602)

Inspect or clean up the working tree at /var/folders/j2/xwvw8qrn6pj3j_hz29rnr66r0000gn/T/python-build.20170914090545.38590
Results logged to /var/folders/j2/xwvw8qrn6pj3j_hz29rnr66r0000gn/T/python-build.20170914090545.38590.log

Last 10 log lines:
  File "/private/var/folders/j2/xwvw8qrn6pj3j_hz29rnr66r0000gn/T/python-build.20170914090545.38590/Python-3.5.0/Lib/ensurepip/__main__.py", line 4, in <module>
    ensurepip._main()
  File "/private/var/folders/j2/xwvw8qrn6pj3j_hz29rnr66r0000gn/T/python-build.20170914090545.38590/Python-3.5.0/Lib/ensurepip/__init__.py", line 209, in _main
    default_pip=args.default_pip,
  File "/private/var/folders/j2/xwvw8qrn6pj3j_hz29rnr66r0000gn/T/python-build.20170914090545.38590/Python-3.5.0/Lib/ensurepip/__init__.py", line 116, in bootstrap
    _run_pip(args + [p[0] for p in _PROJECTS], additional_paths)
  File "/private/var/folders/j2/xwvw8qrn6pj3j_hz29rnr66r0000gn/T/python-build.20170914090545.38590/Python-3.5.0/Lib/ensurepip/__init__.py", line 40, in _run_pip
    import pip
zipimport.ZipImportError: can't decompress data; zlib not available
make: *** [install] Error 1
pyenv: version `3.5.0' not installed
/Applications/AYAB-Launcher.app/Contents/Resources/script: line 50: /Users/VJ/.pyenv/shims/pip3: No such file or directory
/Applications/AYAB-Launcher.app/Contents/Resources/script: line 51: /Users/VJ/.pyenv/shims/pip3: No such file or directory
/Applications/AYAB-Launcher.app/Contents/Resources/script: line 52: /Users/VJ/.pyenv/shims/pip3: No such file or directory
/Applications/AYAB-Launcher.app/Contents/Resources/script: line 53: /Users/VJ/.pyenv/shims/pip3: No such file or directory
/Applications/AYAB-Launcher.app/Contents/Resources/script: line 58: /Users/VJ/.pyenv/shims/python3: No such file or directory
```

@skateness Can you give us the exact specs of your computer? Which exact Mac Hardware and which exact macOS version do you use?

**Comment by derchrisuk** [comment]
Can you also ZIP the content of `/var/folders/j2/xwvw8qrn6pj3j_hz29rnr66r0000gn/T/python-build.20170914090545.38590` and get the build log `/var/folders/j2/xwvw8qrn6pj3j_hz29rnr66r0000gn/T/python-build.20170914090545.38590.log`

**Comment by derchrisuk** [comment]
Can you also run the following in a terminal:

    $ xcode-select --install

**Comment by skateness** [comment]
My hardware / software specs: MacBook Pro (Retina, 13 inch, Late 2013); Mac OS Sierra Version 10.12.6; 2.8 Ghz Intel Core i7; 8 GB 1600 Mhz DDR3 memory; Intel Iris 1536MB graphics

**Comment by skateness** [comment]
[readline-6.3.zip](https://github.com/AllYarnsAreBeautiful/ayab-desktop/files/1306979/readline-6.3.zip)

**Comment by skateness** [comment]
Readline folder uploaded.  Python folder is too big, even when zipped.  Let me know if you'd like me to zip and sort the files within that folder.

**Comment by skateness** [comment]
Contents of /var/folders/j2/xwvw8qrn6pj3j_hz29rnr66r0000gn/T/python-build.20170914090545.38590.log :

Downloading readline-6.3.tar.gz...
-> https://pyenv.github.io/pythons/56ba6071b9462f980c5a72ab0023893b65ba6debb4eeb475d7a563dc65cafd43
Installing readline-6.3...
Installed readline-6.3 to /Users/VJ/.pyenv/versions/3.5.0

Downloading Python-3.5.0.tar.xz...
-> https://www.python.org/ftp/python/3.5.0/Python-3.5.0.tar.xz
Installing Python-3.5.0...

BUILD FAILED (OS X 10.12.6 using python-build 20160602)

Inspect or clean up the working tree at /var/folders/j2/xwvw8qrn6pj3j_hz29rnr66r0000gn/T/python-build.20170914090545.38590
Results logged to /var/folders/j2/xwvw8qrn6pj3j_hz29rnr66r0000gn/T/python-build.20170914090545.38590.log

Last 10 log lines:
  File "/private/var/folders/j2/xwvw8qrn6pj3j_hz29rnr66r0000gn/T/python-build.20170914090545.38590/Python-3.5.0/Lib/ensurepip/__main__.py", line 4, in <module>
    ensurepip._main()
  File "/private/var/folders/j2/xwvw8qrn6pj3j_hz29rnr66r0000gn/T/python-build.20170914090545.38590/Python-3.5.0/Lib/ensurepip/__init__.py", line 209, in _main
    default_pip=args.default_pip,
  File "/private/var/folders/j2/xwvw8qrn6pj3j_hz29rnr66r0000gn/T/python-build.20170914090545.38590/Python-3.5.0/Lib/ensurepip/__init__.py", line 116, in bootstrap
    _run_pip(args + [p[0] for p in _PROJECTS], additional_paths)
  File "/private/var/folders/j2/xwvw8qrn6pj3j_hz29rnr66r0000gn/T/python-build.20170914090545.38590/Python-3.5.0/Lib/ensurepip/__init__.py", line 40, in _run_pip
    import pip
zipimport.ZipImportError: can't decompress data; zlib not available
make: *** [install] Error 1
pyenv: version `3.5.0' not installed
/Applications/AYAB-Launcher.app/Contents/Resources/script: line 50: /Users/VJ/.pyenv/shims/pip3: No such file or directory
/Applications/AYAB-Launcher.app/Contents/Resources/script: line 51: /Users/VJ/.pyenv/shims/pip3: No such file or directory
/Applications/AYAB-Launcher.app/Contents/Resources/script: line 52: /Users/VJ/.pyenv/shims/pip3: No such file or directory
/Applications/AYAB-Launcher.app/Contents/Resources/script: line 53: /Users/VJ/.pyenv/shims/pip3: No such file or directory
/Applications/AYAB-Launcher.app/Contents/Resources/script: line 58: /Users/VJ/.pyenv/shims/python3: No such file or directory

**Comment by skateness** [comment]
I've run the command but get this response (perhaps I'm doing something wrong?)

Last login: Fri Sep 15 12:36:06 on ttys000
Vanessas-MacBook-Pro:~ VJ$ $ xcode-select --install
-bash: $: command not found
Vanessas-MacBook-Pro:~ VJ$ $ xcode-select --install
-bash: $: command not found
Vanessas-MacBook-Pro:~ VJ$

**Comment by derchrisuk** [comment]
From the log Xcode tools were already installed, so the above command should be available and return some output.
As for the error during the installation of python, I can see where it is failing, but not sure why.
It is in one of the required components to get python installed.
I would suggest to re-run the install.
To do this, delete the following folder: `/Users/VJ/.pyenv`
And start the AYAB Launcher again.
It should start from scratch getting pyenv and python installed.

**Comment by skateness** [comment]
I was able to run and install the developer tools.  I deleted the /Users/VJ/.pyenv folder and started the AYAB Launcher again.  This time it loaded!  Thank you.  I'm going to use the software now to confirm that all is well.  Much appreciated!

**Comment by fafa2292** [comment]
Hi guys,
I've got a similar problem but here's what I have 
/Users/fafa2292/Desktop/AYAB PB/1.1.png
/Users/fafa2292/Desktop/AYAB PB/2.png
/Users/fafa2292/Desktop/AYAB PB/3.png

Also, I have Mac OSX 10.9.5

**Comment by fafa2292** [comment]
Hi guys,
I have Mac OSX 10.9.5 and I've got a similar problem but here's what I have

**Comment by fafa2292** [comment]
Hi guys,
I have Mac OSX 10.9.5 and I've got a similar problem but here's what I have

**Comment by fafa2292** [comment]
Hi guys,
I have Mac OSX 10.9.5 and I've got a similar problem but here's what I have

**Comment by fafa2292** [comment]
Hi guys,
I have Mac OSX 10.9.5 and I've got a similar problem but here's what I have :
![1 1](https://user-images.githubusercontent.com/36538109/36310344-fb5dc378-1327-11e8-8dec-9476bfd57f8f.png)
![2](https://user-images.githubusercontent.com/36538109/36310345-fb7d2ccc-1327-11e8-932d-75af15b559e1.png)
![3](https://user-images.githubusercontent.com/36538109/36310346-fb939408-1327-11e8-8ba2-976ef7baee70.png)

**Comment by derchrisuk** [comment]
Can you open a terminal and run the following:

```
wget https://gist.github.com/derchrisuk/cf45099b5d98b8f9f6a311e4fcc95ec2/raw/e088ac844e695b9b13c14d0951c19a86cdc8e238/qt5ver.py
~/.pyenv/shims/python3 qt5ver.py
```

And post the output here

**Comment by fafa2292** [comment]
Ok now It’s okay , I have the update of my mac and Ayab opens normally.

The problem now is that he can not find the port. How to do? Knowing that I have a mac…

Thanks :)





Le 18 févr. 2018 à 17:05, Christian Gerbrandt <notifications@github.com<mailto:notifications@github.com>> a écrit :


Can you open a terminal and run the following:

wget https://gist.github.com/derchrisuk/cf45099b5d98b8f9f6a311e4fcc95ec2/raw/e088ac844e695b9b13c14d0951c19a86cdc8e238/qt5ver.py
~/.pyenv/shims/python3 qt5ver.py


And post the output here

**Comment by oskay** [comment]
You most likely need to install the USB driver now.

**Comment by fafa2292** [comment]
What USB driver? (btw I have AYAB interface)



Le 20 févr. 2018 à 19:55, Windell Oskay <notifications@github.com<mailto:notifications@github.com>> a écrit :


You most likely need to install the USB driver now.

**Comment by oskay** [comment]
For most boards, the SiLabs CP210x "VCP" Drivers.

**Comment by chris007de** [comment]
Should be fixed with #253
